eslint-plugin-expensify/prefer-type-fest.js

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable rulesdir/prefer-underscore-method */
+/* eslint-disable es/no-optional-chaining */
+const {AST_NODE_TYPES} = require('@typescript-eslint/utils');
+const {PREFER_TYPE_FEST_VALUE_OF, PREFER_TYPE_FEST_TUPLE_TO_UNION} = require('./CONST').MESSAGE;
+
+const rule = {
+    meta: {
+        type: 'suggestion',
+        docs: {
+            description: 'Enforce using type-fest utility types for better type readability',
+        },
+        fixable: 'code',
+    },
+    create(context) {
+        let typeFestImported = false;
+
+        function valueOfFixer(node, objectName) {
+            return (fixer) => {
+                // Create replacements and add import if necessary
+                const fixes = [fixer.replaceText(node, `ValueOf<typeof ${objectName}>`)];
+    
+                if (!typeFestImported) {
+                    fixes.push(
+                        fixer.insertTextBefore(
+                            context.getSourceCode().ast.body[0],
+                            "import type {ValueOf} from 'type-fest';\n"
+                        )
```

Comment:
Maybe you could extract this into a separate function (addImport), so can be used in the other use case and more in the future if we have anothers?